adapters/repos/db/index.go

```diff
@@ -340,6 +335,12 @@ func (i *Index) determineObjectShard(id strfmt.UUID, tenantKey string, ss *shard
 			return "", fmt.Errorf("cannot find sharding state for class %q", i.Config.ClassName.String())
 		}
 	}
+	if ss.PartitioningEnabled && tenantKey == "" {
+		return "", errors.New("object without tenantName for class with multi tenancy")
+	} else if !ss.PartitioningEnabled && tenantKey != "" {
+		return "", errors.New("object with tenantName for class without multi tenancy")
```

Comment:
I would define both errors (errors.New()) outside the function so that user can check the specific errors.

Reply:
errors to be refactored in separate PR